src/quickProvider.ts

```diff
@@ -95,69 +92,72 @@ export class QuickActionProvider implements vscode.CodeActionProvider {
-                middle_line_of_selection = cursorPosition.line;
-            }
+        // if no diagnostic were present, taking user's selection instead
+        if (actionId === 'bugs' && context?.diagnostics && context.diagnostics.length > 0) {
+            const diagnostic = context.diagnostics[0];
+            diagnosticMessage = diagnostic.message;
+            middleLineOfSelection = diagnostic.range.start.line;
+
+            codeSnippet = editor.document.getText(diagnostic.range);
         } else {
-            code_snippet = editor.document.getText(selection);
-            middle_line_of_selection = Math.floor((selection.start.line + selection.end.line) / 2);
+            codeSnippet = editor.document.getText(selection);
+            middleLineOfSelection = Math.floor((selection.start.line + selection.end.line) / 2);
         }
 
-        const messages: PlainTextMessage[] = command.messages.map(({ content }) => ({
-            role: 'system',
-            content: content
-                .replace("%ARGS%", '')
+        if (!codeSnippet) {
+            const cursorPosition = selection.isEmpty ? editor.selection.active : selection.start;
+            const startLine = Math.max(0, cursorPosition.line - 2);
+            const endLine = Math.min(editor.document.lineCount - 1, cursorPosition.line + 2);
+            const range = new vscode.Range(startLine, 0, endLine, editor.document.lineAt(endLine).text.length);
+            codeSnippet = editor.document.getText(range);
+            middleLineOfSelection = cursorPosition.line;
+        }
+
+        const messageBlock = command.messages.map(({content}) => (
+            content
+                // we should fetch default prompt somehow
+                .replace("%PROMPT_DEFAULT%", '')
                 .replace("%CURRENT_FILE_PATH_COLON_CURSOR%", '')
-                .replace("%CURRENT_FILE%", working_on_attach_filename)
-                .replace("%CURSOR_LINE%", (middle_line_of_selection + 1).toString())
-                .replace("%CODE_SELECTION%", code_snippet)
-        }));
-
-        const question = actionId === 'bugs' ? (diagnostic_message || 'Issue with code') : 'Issue';
-
-        const chat: ChatThread = {
-            id: uuidv4(),
-            title: question,
-            messages: [
-                ...messages,
-                { role: 'system', content: `Active file: ${working_on_attach_filename}` },
-                ...(actionId === 'bugs' && diagnostic_message
-                    ? [{ role: 'user', content: `Error in: ${diagnostic_message}` }]
-                    : [])
-            ] as ChatMessage[],
-            model: '',
-        };
+                .replace("%CURRENT_FILE%", filePath)
+                .replace("%CURSOR_LINE%", (middleLineOfSelection + 1).toString())
+                .replace("%CODE_SELECTION%", codeSnippet + "\n")
+        )).join("\n");
+
         vscode.commands.executeCommand("refactaicmd.callChat");
-        global.side_panel?.goto_chat(chat);
```

Comment:
I think this may have caused a regression when chat hasn't been started,

Is there a way to check if chat is open? and if not set the initial state, wait (maybe send a ready message), then add the text?

Reply:
`goto_chat()` function is handling complete cycle of chat opening and sending prepared chat thread inside, so we cannot control how it behaves. Other way around is to save all necessary data from frontend inside of vscode PanelWebview` class, but in that case we'll be doing stuff not flexibly. But we can implement a special check for if chat was not opened yet, make a closure and then `sendQuickActionToChat` call

Reply:
But therefore we'll encounter an issue with opening chat twice, the same as clicking on CodeLens before chat initialization.